hledger-lib/Hledger/Read/RulesReader.hs

```diff
@@ -1596,9 +1596,14 @@ tests_RulesReader = testGroup "RulesReader" [
    --    parseWithState' defrules matcherp "%description ~ A A\n" @?= (Right $ FieldMatcher "%description" "A A")
    ]
 
+
   ,testGroup "regexp" [
     testCase "regexp.ends-before-&&" $
-      parseWithState' defrules (regexp empty) "A A && xxx" @?= (Right $ toRegexCI' "A A")
+      parseWithState' defrules (regexp eof) "A A && xxx" @?= (Right $ toRegexCI' "A A")
+   ,testCase "regexp contains &" $
+      parseWithState' defrules (regexp eof) "A & B" @?= (Right $ toRegexCI' "A & B")
+   ,testCase "regexp contains escaped &" $
+      parseWithState' defrules (regexp eof) "A \\& B" @?= (Right $ toRegexCI' "A \\& B")
```

Comment:
I don't think we support any backslash escaping here, so no need to test for it here or in csv.test.

Reply:
Do you want to support escaping double ampersands?

2025-01-01,COFFEE && CODE,100
2025-01-02,COFFEE CODE,200

if
%description COFFEE \&\& CODE
    account2 expenses:bars

That's currently working: the rule only gets applied to the first line. But it's not documented nor tested.

Reply:
I'm releasing right now and we didn't discuss it, so I'll pass. That could use a bit more thinking about.